tests/unit/test_cursorManager.py

```diff
@@ -0,0 +1,78 @@
+#tests/unit/test_cursorManager.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited
+
+"""Unit tests for the cursorManager module.
+"""
+
+import unittest
+import cursorManager
+from . import TestCase
+from .textProvider import BasicTextProvider
+
+class CursorManager(cursorManager.CursorManager, BasicTextProvider):
+	"""CursorManager which navigates within a provided string of text.
+	"""
+
+class TestMove(TestCase):
+
+	def test_nextChar(self):
+		cm = CursorManager(text="abc") # Caret at "a"
+		cm.script_moveByCharacter_forward(None)
+		self.assertEqual(cm.selectionOffsets, (1, 1)) # Caret at "b"
+
+	def test_prevChar(self):
+		cm = CursorManager(text="abc", selection=(1, 1)) # Caret at "b"
+		cm.script_moveByCharacter_back(None)
+		self.assertEqual(cm.selectionOffsets, (0, 0)) # Caret at "a"
+
+class TestSelection(TestCase):
+
+	def test_selectNextChar(self):
+		cm = CursorManager(text="abc") # Caret at "a"
+		cm.script_selectCharacter_forward(None)
+		self.assertEqual(cm.selectionOffsets, (0, 1)) # "a" selected
+
+	def test_selectPrevChar(self):
+		cm = CursorManager(text="abc", selection=(1, 1)) # Caret at "b"
+		cm.script_selectCharacter_back(None)
+		self.assertEqual(cm.selectionOffsets, (0, 1)) # "a" selected
+
+	def test_unselectPrevChar(self):
+		cm = CursorManager(text="abc") # Caret at "a"
+		cm.script_selectCharacter_forward(None) # "a" selected
+		cm.script_selectCharacter_back(None) # "a" unselected
+		self.assertEqual(cm.selectionOffsets, (0, 0)) # Caret at "a", no selection
+
+	def test_selForwardThenUnselThenSelBackward(self):
+		"""Test selecting forward, then unselecting and selecting backward.

```

Comment:
The docstring could be more explicit about the dependencies between tests / behaviour.